<a href="https://colab.research.google.com/github/Tanisha2054/DV-Task1/blob/main/dvtask2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
import pandas as pd
import numpy as np
import seaborn as sns

In [11]:
df=pd.read_csv('/content/tested.csv')

In [12]:
df.head()

PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   

    Ticket     Fare Cabin Embarked  
0   330911   7.8292   NaN        Q  
1   363272   7.0000   NaN        S  
2   240276   9.6875   NaN        Q  
3   315154   8.6625   NaN        S  
4  3101298  12.2875   NaN        S

In [13]:
df.tail()

PassengerId  Survived  Pclass                          Name     Sex  \
413         1305         0       3            Spector, Mr. Woolf    male   
414         1306         1       1  Oliva y Ocana, Dona. Fermina  female   
415         1307         0       3  Saether, Mr. Simon Sivertsen    male   
416         1308         0       3           Ware, Mr. Frederick    male   
417         1309         0       3      Peter, Master. Michael J    male   

      Age  SibSp  Parch              Ticket      Fare Cabin Embarked  
413   NaN      0      0           A.5. 3236    8.0500   NaN        S  
414  39.0      0      0            PC 17758  108.9000  C105        C  
415  38.5      0      0  SOTON/O.Q. 3101262    7.2500   NaN        S  
416   NaN      0      0              359309    8.0500   NaN        S  
417   NaN      1      1                2668   22.3583   NaN        C

In [14]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 418 entries, 0 to 417
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  418 non-null    int64  
 1   Survived     418 non-null    int64  
 2   Pclass       418 non-null    int64  
 3   Name         418 non-null    object 
 4   Sex          418 non-null    object 
 5   Age          332 non-null    float64
 6   SibSp        418 non-null    int64  
 7   Parch        418 non-null    int64  
 8   Ticket       418 non-null    object 
 9   Fare         417 non-null    float64
 10  Cabin        91 non-null     object 
 11  Embarked     418 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 39.3+ KB


In [15]:
df.head(10)

PassengerId  Survived  Pclass  \
0          892         0       3   
1          893         1       3   
2          894         0       2   
3          895         0       3   
4          896         1       3   
5          897         0       3   
6          898         1       3   
7          899         0       2   
8          900         1       3   
9          901         0       3   

                                           Name     Sex   Age  SibSp  Parch  \
0                              Kelly, Mr. James    male  34.5      0      0   
1              Wilkes, Mrs. James (Ellen Needs)  female  47.0      1      0   
2                     Myles, Mr. Thomas Francis    male  62.0      0      0   
3                              Wirz, Mr. Albert    male  27.0      0      0   
4  Hirvonen, Mrs. Alexander (Helga E Lindqvist)  female  22.0      1      1   
5                    Svensson, Mr. Johan Cervin    male  14.0      0      0   
6                          Connolly, Miss. Kate  female  30.0      0      0   
7                  Caldwell, Mr. Albert Francis    male  26.0      1      1   
8     Abrahim, Mrs. Joseph (Sophie Halaut Easu)  female  18.0      0      0   
9                       Davies, Mr. John Samuel    male  21.0      2      0   

      Ticket     Fare Cabin Embarked  
0     330911   7.8292   NaN        Q  
1     363272   7.0000   NaN        S  
2     240276   9.6875   NaN        Q  
3     315154   8.6625   NaN        S  
4    3101298  12.2875   NaN        S  
5       7538   9.2250   NaN        S  
6     330972   7.6292   NaN        Q  
7     248738  29.0000   NaN        S  
8       2657   7.2292   NaN        C  
9  A/4 48871  24.1500   NaN        S

In [16]:
df.tail(10)

PassengerId  Survived  Pclass  \
408         1300         1       3   
409         1301         1       3   
410         1302         1       3   
411         1303         1       1   
412         1304         1       3   
413         1305         0       3   
414         1306         1       1   
415         1307         0       3   
416         1308         0       3   
417         1309         0       3   

                                                Name     Sex   Age  SibSp  \
408                  Riordan, Miss. Johanna Hannah""  female   NaN      0   
409                        Peacock, Miss. Treasteall  female   3.0      1   
410                           Naughton, Miss. Hannah  female   NaN      0   
411  Minahan, Mrs. William Edward (Lillian E Thorpe)  female  37.0      1   
412                   Henriksson, Miss. Jenny Lovisa  female  28.0      0   
413                               Spector, Mr. Woolf    male   NaN      0   
414                     Oliva y Ocana, Dona. Fermina  female  39.0      0   
415                     Saether, Mr. Simon Sivertsen    male  38.5      0   
416                              Ware, Mr. Frederick    male   NaN      0   
417                         Peter, Master. Michael J    male   NaN      1   

     Parch              Ticket      Fare Cabin Embarked  
408      0              334915    7.7208   NaN        Q  
409      1  SOTON/O.Q. 3101315   13.7750   NaN        S  
410      0              365237    7.7500   NaN        Q  
411      0               19928   90.0000   C78        Q  
412      0              347086    7.7750   NaN        S  
413      0           A.5. 3236    8.0500   NaN        S  
414      0            PC 17758  108.9000  C105        C  
415      0  SOTON/O.Q. 3101262    7.2500   NaN        S  
416      0              359309    8.0500   NaN        S  
417      1                2668   22.3583   NaN        C

In [17]:
df.dropna(
)

PassengerId  Survived  Pclass  \
12           904         1       1   
14           906         1       1   
24           916         1       1   
26           918         1       1   
28           920         0       1   
..           ...       ...     ...   
404         1296         0       1   
405         1297         0       2   
407         1299         0       1   
411         1303         1       1   
414         1306         1       1   

                                                  Name     Sex   Age  SibSp  \
12       Snyder, Mrs. John Pillsbury (Nelle Stevenson)  female  23.0      1   
14   Chaffee, Mrs. Herbert Fuller (Carrie Constance...  female  47.0      1   
24     Ryerson, Mrs. Arthur Larned (Emily Maria Borie)  female  48.0      1   
26                        Ostby, Miss. Helene Ragnhild  female  22.0      0   
28                             Brady, Mr. John Bertram    male  41.0      0   
..                                                 ...     ...   ...    ...   
404                       Frauenthal, Mr. Isaac Gerald    male  43.0      1   
405       Nourney, Mr. Alfred (Baron von Drachstedt")"    male  20.0      0   
407                         Widener, Mr. George Dunton    male  50.0      1   
411    Minahan, Mrs. William Edward (Lillian E Thorpe)  female  37.0      1   
414                       Oliva y Ocana, Dona. Fermina  female  39.0      0   

     Parch         Ticket      Fare            Cabin Embarked  
12       0          21228   82.2667              B45        S  
14       0    W.E.P. 5734   61.1750              E31        S  
24       3       PC 17608  262.3750  B57 B59 B63 B66        C  
26       1         113509   61.9792              B36        C  
28       0         113054   30.5000              A21        S  
..     ...            ...       ...              ...      ...  
404      0          17765   27.7208              D40        C  
405      0  SC/PARIS 2166   13.8625              D38        C  
407      1         113503  211.5000              C80        C  
411      0          19928   90.0000              C78        Q  
414      0       PC 17758  108.9000             C105        C  

[87 rows x 12 columns]

In [18]:
df.dropna(inplace = True)

print(df.to_string())

     PassengerId  Survived  Pclass                                                             Name     Sex   Age  SibSp  Parch         Ticket      Fare            Cabin Embarked
12           904         1       1                    Snyder, Mrs. John Pillsbury (Nelle Stevenson)  female  23.0      1      0          21228   82.2667              B45        S
14           906         1       1          Chaffee, Mrs. Herbert Fuller (Carrie Constance Toogood)  female  47.0      1      0    W.E.P. 5734   61.1750              E31        S
24           916         1       1                  Ryerson, Mrs. Arthur Larned (Emily Maria Borie)  female  48.0      1      3       PC 17608  262.3750  B57 B59 B63 B66        C
26           918         1       1                                     Ostby, Miss. Helene Ragnhild  female  22.0      0      1         113509   61.9792              B36        C
28           920         0       1                                          Brady, Mr. John Bertram    ma

In [19]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 87 entries, 12 to 414
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  87 non-null     int64  
 1   Survived     87 non-null     int64  
 2   Pclass       87 non-null     int64  
 3   Name         87 non-null     object 
 4   Sex          87 non-null     object 
 5   Age          87 non-null     float64
 6   SibSp        87 non-null     int64  
 7   Parch        87 non-null     int64  
 8   Ticket       87 non-null     object 
 9   Fare         87 non-null     float64
 10  Cabin        87 non-null     object 
 11  Embarked     87 non-null     object 
dtypes: float64(2), int64(5), object(5)
memory usage: 8.8+ KB


In [20]:
print(df.duplicated())

12     False
14     False
24     False
26     False
28     False
       ...  
404    False
405    False
407    False
411    False
414    False
Length: 87, dtype: bool


In [21]:
df.drop_duplicates(inplace = True)

In [22]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 87 entries, 12 to 414
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  87 non-null     int64  
 1   Survived     87 non-null     int64  
 2   Pclass       87 non-null     int64  
 3   Name         87 non-null     object 
 4   Sex          87 non-null     object 
 5   Age          87 non-null     float64
 6   SibSp        87 non-null     int64  
 7   Parch        87 non-null     int64  
 8   Ticket       87 non-null     object 
 9   Fare         87 non-null     float64
 10  Cabin        87 non-null     object 
 11  Embarked     87 non-null     object 
dtypes: float64(2), int64(5), object(5)
memory usage: 8.8+ KB


In [23]:
df.corr()

<ipython-input-23-2f6f6606aa2c>:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.058260  0.004934  0.055488 -0.087828 -0.122551   
Survived       -0.058260  1.000000 -0.056699  0.022983  0.170596  0.181554   
Pclass          0.004934 -0.056699  1.000000 -0.410924 -0.132790  0.006411   
Age             0.055488  0.022983 -0.410924  1.000000  0.062530  0.051144   
SibSp          -0.087828  0.170596 -0.132790  0.062530  1.000000  0.252194   
Parch          -0.122551  0.181554  0.006411  0.051144  0.252194  1.000000   
Fare           -0.097346  0.179047 -0.298186  0.180567  0.213014  0.395685   

                 Fare  
PassengerId -0.097346  
Survived     0.179047  
Pclass      -0.298186  
Age          0.180567  
SibSp        0.213014  
Parch        0.395685  
Fare         1.000000

The above is the Titanic dataset analysis.The data is initially displayed without performing any operations. Null values/cloumns from the data have been dropped. Duplicates have been recognised and duplicate columns have been dropped.The dataset initially consisted of 418 entries but after all the cleaning operations the dataset now contains 87 entries.Lastly, the correlation between the columns has been displayed where '1' represents a perfect correlation.